osquery/tables/system/darwin/unified_log.mm

```diff
@@ -0,0 +1,218 @@
+      NSDate* provided_date =
+          [NSDate dateWithTimeIntervalSince1970:provided_timestamp];
+
+      position = [logstore positionWithDate:provided_date];
+    }
+
+    // grab all the supported columns used in simple constraints and make a
+    // compound predicate out of them.
+    NSMutableArray* subpredicates = [[NSMutableArray alloc] init];
+    for (const auto& it : queryContext.constraints) {
+      const std::string& key = it.first;
+      for (const auto& constraint : it.second.getAll()) {
+        addQueryOp(subpredicates,
+                   key,
+                   constraint.expr,
+                   static_cast<ConstraintOperator>(constraint.op));
+      }
+    }
+
+    NSPredicate* predicate =
+        [NSCompoundPredicate andPredicateWithSubpredicates:subpredicates];
+
+    // enumerate the entries in ascending order by timestamp
+    OSLogEnumeratorOptions option = 0;
+
+    OSLogEnumerator* enumerator =
+        [logstore entriesEnumeratorWithOptions:option
+                                      position:position
+                                     predicate:predicate
+                                         error:&error];
+    if (error != nil) {
+      TLOG << "error enumerating entries in system log: "
+           << [[error localizedDescription] UTF8String];
+      return;
+    }
+    for (OSLogEntryLog* entry in enumerator) {
+      Row r;
+      r["timestamp"] = BIGINT([[entry date] timeIntervalSince1970]);
+      r["message"] = TEXT([[entry composedMessage] UTF8String]);
+      r["storage"] = INTEGER([entry storeCategory]);
+
+      if ([entry respondsToSelector:@selector(activityIdentifier)]) {
+        r["activity"] = INTEGER([entry activityIdentifier]);
+        r["process"] = TEXT([[entry process] UTF8String]);
+        r["pid"] = INTEGER([entry processIdentifier]);
+        r["sender"] = TEXT([[entry sender] UTF8String]);
+        r["tid"] = INTEGER([entry threadIdentifier]);
+      }
+
+      if ([entry respondsToSelector:@selector(subsystem)]) {
+        NSString* subsystem = [entry subsystem];
+        if (subsystem != nil) {
+          r["subsystem"] = TEXT([subsystem UTF8String]);
+        }
+        NSString* category = [entry category];
+        if (category != nil) {
+          r["category"] = TEXT([category UTF8String]);
+        }
+      }
+      if ([entry respondsToSelector:@selector(level)]) {
+        const char* logLevelNames[] = {
+            [OSLogEntryLogLevelUndefined] = "undefined",
+            [OSLogEntryLogLevelDebug] = "debug",
+            [OSLogEntryLogLevelInfo] = "info",
+            [OSLogEntryLogLevelNotice] = "default",
+            [OSLogEntryLogLevelError] = "error",
+            [OSLogEntryLogLevelFault] = "fault",
+        };
+
+        r["level"] = TEXT(logLevelNames[[entry level]]);
+      }
+      results.push_back(r);
+    }
+  }
+}
+
+QueryData genUnifiedLog(QueryContext& context) {
+  QueryData results;
+  if (@available(macOS 10.15, *)) {
```

Comment:
If `QueryData genUnifiedLog(QueryContext& context)` is just doing a macOS 10.15 check and calling `void genUnifiedLog` which does the same check, are both routines needed?

Reply:
good catch, I've cleaned it up